test/integration/logging_test.go

```diff
@@ -0,0 +1,236 @@
+	}, retry.Attempts(3))
+
+	if retryErr != nil {
+		t.Fatal(retryErr)
+	}
+
+	defer close(portForward)
+
+	_, errNamespace := clientset.CoreV1().Namespaces().Create(context.Background(), &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}, metav1.CreateOptions{})
+	if errNamespace != nil {
+		t.Fatal(errNamespace)
+	}
+	// This register the function.
+	t.Cleanup(func() {
+		// delete the test namespace
+		errNamespace = clientset.CoreV1().Namespaces().Delete(context.Background(), namespaceName, metav1.DeleteOptions{})
+		if errNamespace != nil {
+			t.Fatal(errNamespace)
+		}
+	})
+
+	_, roleBindingErr := clientset.RbacV1().RoleBindings(namespaceName).Create(context.Background(), &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "view-default-test",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "default",
+				Namespace: namespaceName,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "view",
+		},
+	},
+		metav1.CreateOptions{})
+	if roleBindingErr != nil {
+		t.Fatal(roleBindingErr)
+	}
+
+	token, tokenErr := clientset.CoreV1().ServiceAccounts(namespaceName).CreateToken(context.Background(), "default", &authenticationv1.TokenRequest{}, metav1.CreateOptions{})
+	if tokenErr != nil {
+		fmt.Printf("could not create a token, %s", tokenErr)
+		t.Fatal(tokenErr)
+	}
+
+	clientHTTP := http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	kac := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "KubeArchiveConfig",
+			"apiVersion": fmt.Sprintf("%s/%s", kubearchivev1alpha1.SchemeBuilder.GroupVersion.Group, kubearchivev1alpha1.SchemeBuilder.GroupVersion.Version),
+			"metadata": map[string]string{
+				"name":      "kubearchive",
+				"namespace": namespaceName,
+			},
+			"spec": map[string]any{
+				"resources": []map[string]any{
+					{
+						"selector": map[string]string{
+							"apiVersion": "v1",
+							"kind":       "Pod",
+						},
+						"archiveWhen": "true",
+						"deleteWhen":  "status.phase == 'Succeeded'",
+					},
+				},
+			},
+		},
+	}
+
+	gvr := kubearchivev1alpha1.GroupVersion.WithResource("kubearchiveconfigs")
+	_, kacErr := dynamicClient.Resource(gvr).Namespace(namespaceName).Create(context.Background(), kac, metav1.CreateOptions{})
+	if kacErr != nil {
+		t.Fatal(kacErr)
+	}
+
+	// Give a bit of time to KubeArchive to pickup filters changes
+	time.Sleep(5 * time.Second)
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fedora",
+			Namespace: namespaceName,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Containers: []corev1.Container{
+				corev1.Container{
+					Name:    "fedora",
+					Command: []string{"echo", "Hello World"},
+					Image:   "quay.io/fedora/fedora:latest",
+				},
```

Comment:
I'd suggest taking advantage of the image that we are using in the cronjob-generator script for having container with more "realistic" logs than just a one line.

```suggestion
				corev1.Container{
					Name:    "apache-log-generator",
					Command: []string{"echo", "Hello World"},
					Image:   "quay.io/kubearchive/mingrammer/flog",
					ImagePullPolicy: PullPolicy.PullIfNotPresent,
                                       Args: []string{"-n","1024","-d","5ms"}
				},
```

After that, and if we end up only having one pod from where to retrieve the logs, an additional assertion could be to test if the lines retrieved are 1024:

```
	assert.Equal(t, 1024, len(strings.Split(data, "\n"))
```

Reply:
Thanks, I will switch to this one.